platform/android/java/app/config.gradle

```
@@ -382,17 +384,12 @@ ext.shouldNotStrip = { ->
  */
 ext.shouldUseLegacyPackaging = { ->
     int minSdk = getExportMinSdkVersion()
-    if (minSdk < 23) {
-        // Enforce the default behavior for compatibility with device running api < 23
-        return true
-    }
-
     String legacyPackagingFlag = project.hasProperty("compress_native_libraries") ? project.property("compress_native_libraries") : ""
     if (legacyPackagingFlag != null && !legacyPackagingFlag.isEmpty()) {
         return Boolean.parseBoolean(legacyPackagingFlag)
     }
 
-    // Default behavior for minSdk >= 23
+    // Default behavior for minSdk >= 24
     return false
```

```suggestion
    return true
```

I was only able to test successfully with "true" (the APK is also smaller afterwards)

---

@Alex2782 Setting this to `true` turns on legacy packaging, which while it makes the apk smaller, it increases install size because a copy of the libraries is now required by the system.

See https://developer.android.com/build/releases/past-releases/agp-3-6-0-release-notes#extractNativeLibs and https://stackoverflow.com/a/44704840 for more details.

I'll take a look at the templates not installing properly.

---

Can you reproduce it? Otherwise, I'll look further later.


I tried out the recent changes and did some quick research. The exported APK looks correct to me, and my device (Samsung Tab S7) should support it as well.

```
adb install Bunnymark.apk

Performing Streamed Install
adb: failed to install Bunnymark.apk: Failure [INSTALL_FAILED_INVALID_APK: INSTALL_FAILED_INVALID_APK: Failed to extract native libraries, res=-2]

------------

unzip -lv Bunnymark.apk | grep "\.so"

 1292904  Stored  1292904   0% 05-08-2025 14:16 51e3dcdc  lib/arm64-v8a/libc++_shared.so
75536960  Stored 75536960   0% 05-08-2025 14:16 e88faaa3  lib/arm64-v8a/libgodot_android.so
```

---

> Can you reproduce it? Otherwise, I'll look further later.

Yeah I was able to reproduce it and I have a fix for the install issue.

The issue was caused by our version of zipalign not properly aligning the uncompressed .so libraries. I'll update the PR soon.

---

@Alex2782 I've update the PR with the install fix, can you take another look.